### PR TITLE
Add a feature flag to use URLSession in WordPressKit

### PR DESCRIPTION
**Pull request by crazytonyli**
## Description

What says in the title. By default, WordPressKit sends HTTP requests using Alamofire. This PR adds a feature flag to configure WordPressKit sends HTTP requests using URLSession for all non-production builds.

With the new feature flag, we can test the new implementation during development. After the testing is done, we can switch WordPressKit over to URLSession completely and remove the new code in this PR.

## Testing instructions

Try features that involves different kinds of HTTP requests (GET, POST, upload files, etc). Testing error cases would be super valuable too.

## Screenshots

N/A

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -87,8 +87,10 @@ target 'WooCommerce' do
   # To allow pod to pick up beta versions use -beta. E.g., 1.1.7-beta.1
   # pod 'WordPressAuthenticator', '~> 9.0'
   # pod 'WordPressAuthenticator', git: 'https://github.com/wordpress-mobile/WordPressAuthenticator-iOS.git', branch: ''
-  pod 'WordPressAuthenticator', git: 'https://github.com/wordpress-mobile/WordPressAuthenticator-iOS.git', commit: '6312d0a91e709a4bae017e04f70e520199928928'
+  pod 'WordPressAuthenticator', git: 'https://github.com/wordpress-mobile/WordPressAuthenticator-iOS.git', commit: 'fa06fca7178b268d382d91861752b3be0729e8a8'
   # pod 'WordPressAuthenticator', path: '../WordPressAuthenticator-iOS'
+
```
Comment (dangermattic) [inline Podfile:92]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/woocommerce/woocommerce-ios/blob/1a810061b98adc999ff68c6508e925265ad8f092/Podfile#L92"></span>Layout/TrailingWhitespace: Trailing whitespace detected.</td>
    </tr>
  </tbody>
</table>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Podfile [hunk 2]
```diff
@@ -87,8 +87,10 @@ target 'WooCommerce' do
...
```
Comment (dangermattic) [inline Podfile]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/woocommerce/woocommerce-ios/blob/02efdaf46c9b082622f3dfc19a087039e534765a/Podfile#L92"></span>Layout/TrailingWhitespace: Trailing whitespace detected.</td>
    </tr>
  </tbody>
</table>

File: WooCommerce/Classes/AppDelegate.swift [hunk 3]
```diff
@@ -102,6 +103,8 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         setupUniversalLinkRouter()
         disableAnimationsIfNeeded()
 
+        configureWordPressKit()
```
Comment (mokagio) [inline WooCommerce/Classes/AppDelegate.swift:106]: I like how this is a function and not an inline call of the flag check + conditional configuration logic 😄

File: Experiments/Experiments/DefaultFeatureFlagService.swift [hunk 4]
```diff
@@ -91,6 +91,8 @@ public struct DefaultFeatureFlagService: FeatureFlagService {
             return buildConfig == .localDeveloper || buildConfig == .alpha
         case .splitViewInProductsTab:
             return false
+        case .useURLSessionInWordPressKit:
+            return buildConfig != .appStore
```
Comment (mokagio) [inline Experiments/Experiments/DefaultFeatureFlagService.swift:97]: Did you consider making this a remote flag?

**Review by mokagio — APPROVED** [review]

File: Experiments/Experiments/DefaultFeatureFlagService.swift (hunk 4 above)
Comment (crazytonyli) [reply Experiments/Experiments/DefaultFeatureFlagService.swift:97]: No, that didn't cross my mind. I don't see the new URLSession thing as a feature that we'd like to remotely turn on or off though.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/185">17.3</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Experiments/Experiments/DefaultFeatureFlagService.swift (hunk 4 above)
Comment (mokagio) [reply Experiments/Experiments/DefaultFeatureFlagService.swift:97]: Good point. The only advantage I see would be in the short transition phase, if we find something tragically wrong in production. But... given how deep and shared the `URLSession` usages are, it seems unlikely something would go unnoticed during testing.

**Pull request merged**